### PR TITLE
[Silabs] Add support for the Lit icd sample app

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Add Silabs Thread support for the Lit ICD sample app

Note for reviewers :

- Documentation might not be completly up to date. Rework of the documentation is underway.
- Apptask does not trigger any LCD updates. This will be added in a follow up PR.

### CODE REVIEW

File: examples/lit-icd-app/silabs/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2020 Project CHIP Authors
```
Comment (yunhanw-google) [inline examples/lit-icd-app/silabs/BUILD.gn:1]: 2023

**Review by yunhanw-google — APPROVED** [review]

File: examples/lit-icd-app/silabs/build_for_wifi_args.gni [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+import("//build_overrides/chip.gni")
+import("${chip_root}/config/standalone/args.gni")
+
+silabs_sdk_target = get_label_info(":sdk", "label_no_toolchain")
+chip_enable_openthread = false
+import("${chip_root}/src/platform/silabs/wifi_args.gni")
+
+chip_enable_ota_requestor = true
+app_data_model = "${chip_root}/examples/light-switch-app/light-switch-common"
```
Comment (jmartinez-silabs) [inline examples/lit-icd-app/silabs/build_for_wifi_args.gni:22]: do you mean to reuse the light-switch datamodel?

File: examples/lit-icd-app/silabs/src/ZclCallbacks.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * @param endpoint   Ver.: always
+ *
+ * TODO Issue #3841
+ * emberAfOnOffClusterInitCallback happens before the stack initialize the cluster
+ * attributes to the default value.
+ * The logic here expects something similar to the deprecated Plugins callback
+ * emberAfPluginOnOffClusterServerPostInitCallback.
+ *
+ */
+void emberAfOnOffClusterInitCallback(EndpointId endpoint)
```
Comment (jmartinez-silabs) [inline examples/lit-icd-app/silabs/src/ZclCallbacks.cpp:62]: you do not need this callback

File: examples/lit-icd-app/silabs/src/AppTask.cpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            sAppTask.DispatchEvent(&event);
+            eventReceived = xQueueReceive(sAppEventQueue, &event, 0);
+        }
+    }
+}
+
+void AppTask::UserActionEventHandler(AppEvent * aEvent)
+{
+    VerifyOrReturn(aEvent->Type == AppEvent::kEventType_Button);
+    // Trigger User Active Mode Event
```
[the remark this answers is not in the document]
Comment (jmartinez-silabs) [reply examples/lit-icd-app/silabs/src/AppTask.cpp:126]: doesn't pb0 on baseApplication already trigger a User Action?

File: examples/lit-icd-app/silabs/openthread.gni [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+import("${chip_root}/config/standalone/args.gni")
+import("${chip_root}/src/platform/silabs/efr32/args.gni")
+
+silabs_sdk_target = get_label_info(":sdk", "label_no_toolchain")
+
+app_data_model = "${chip_root}/examples/light-switch-app/light-switch-common"
```
Comment (jmartinez-silabs) [inline examples/lit-icd-app/silabs/openthread.gni:21]: same comment, why do we use light-switch datamodel

File: examples/lit-icd-app/silabs/src/AppTask.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply examples/lit-icd-app/silabs/src/AppTask.cpp:126]: Renamed the comment. I want to add other user active mode trigger actions. 
Not sure which for now

File: examples/lit-icd-app/silabs/build_for_wifi_args.gni (hunk 2 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/silabs/build_for_wifi_args.gni:22]: done

File: examples/lit-icd-app/silabs/src/ZclCallbacks.cpp (hunk 3 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/silabs/src/ZclCallbacks.cpp]: done

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**